Blog post summary of my last paper

mbonsma opened the pull request:
Here's a blog post summarizing a result I like from my last paper in a (hopefully) very broad and accessible way. I served it locally and it looks good on my end.

mbonsma committed:
draft of null models blog post

mbonsma committed:
images for null models post

mbonsma committed:
add conclusion, small fix to Heler2019 stuff

mbonsma committed:
fix image links

mbonsma committed:
change header image, change figure sizes

mbonsma committed:
fix latex formatting error

mbonsma committed:
small formatting tweaks

mbonsma committed:
remove quotes in subheader - not formatted properly

mbonsma committed:
remove quotes in subheader - not formatted properly

jbRothschild commented:
This is great Maddy! Being familiar with your work, I feel like I may not be the best to be reviewing this to give you feedback for the 'blog'. As such, I'm going to assign it to @ejolysmith . Is that alright Euan?

ejolysmith commented:
I'm happy to! I'll send the feedback within the next few days.

mattsmart commented:
Technical note: not sure if SVG for images will cause issues online (despite working locally). Anybody know? We can merge regardless to test.

ejolysmith commented:
Side note to Matt's comment: when I go to the "file changed" tab on top ^, I can see the png images but not the svg images. It's probably best to avoid svg images in the future so that the reviewers can quickly see the images while they are making comments.

ejolysmith commented on _blog/why-null-models-crispr.md:
```diff
@@ -0,0 +1,83 @@
+
+## Why do we need theory?
+
+Experimentalists wade through the murky waters of biology, hoping to tease out knowledge by controlling the things they can control and carefully observing what they cannot. In a well-designed experiment, the observed data reveals some truth, and this truth is reached by painstakingly ruling out other explanations. This might sound logical and sytematic, but it's an art form: how do you know when ...
```
First sentence is great! I think that the second sentence can be a bit more precise though. I also think it can summarize the main message of your blog: there are often many possible explanations for a given dataset, and theory can help identify them or suss them out. 

So for example, maybe your second sentence could go something like this: In a well-designed experiment, the observed data reveals insight by agreeing with a possible explanation and ruling out others. Or something like that (you can probably make it more elegant haha). I guess the sentence as is all ready says something like this, but when I read _truth_ I don’t know if you mean absolute truth or “truth”. It was only after re-reading the blog that I realized that the “truth” here is a possible explanation, but you’re trying to make the point that there could be others. 

You could also drop “but it’s an art form:”, but that’s just a personal taste.

ejolysmith commented on _blog/why-null-models-crispr.md:
```diff
@@ -0,0 +1,83 @@
+
+Experimentalists wade through the murky waters of biology, hoping to tease out knowledge by controlling the things they can control and carefully observing what they cannot. In a well-designed experiment, the observed data reveals some truth, and this truth is reached by painstakingly ruling out other explanations. This might sound logical and sytematic, but it's an art form: how do you know when ...
+
+This is where theory can help, and in particular, theory so simple that it predicts what would happen if *nothing* weird or cool was happening. Theories like this are often called ''null models'' because they model the ''null'' (''zero'' or ''nothing'') expectation. And if your null model ends up doing something weird or cool after all? Well, you're in business! 
```
I got a little confused here. First paragraph I got: _experiment -> data -> possible explanation, but are there other possible explanations?_ And then here it jumps to: _null models can help_. But why exactly? From what I understand, the goal here is to find other possible explanations for a dataset (or to rule them out), and a first line of investigation would be to try out a null hypothesis which can lead to a prediction using theory. I think something should be added here to clear this up a little.

mbonsma replied:
Thanks! Made some edits to the intro to be more clear.

ejolysmith commented on _blog/why-null-models-crispr.md:
```diff
@@ -0,0 +1,83 @@
+For 60 years since the [1940s](https://en.wikipedia.org/wiki/Luria%E2%80%93Delbr%C3%BCck_experiment), all known bacterial immune processes could be boiled down to *innate* immunity: either you were born lucky with the right mutation, or you die. But in the mid-2000s, [several](https://pubmed.ncbi.nlm.nih.gov/15791728/) [groups](https://pubmed.ncbi.nlm.nih.gov/16079334/) [near](https://pubmed.ncbi ...
+
+Bacteria with a CRISPR immune system can store a memory of past phage infections by cutting out small pieces of phage DNA and inserting these pieces in their own genome. The phage DNA mugshots are (unhelpfully) called *spacers*, and bacteria use them to recognize and bind to the DNA of phages that they've seen before, then neatly snip the phage DNA at that spot[^2], making the phage DNA useless a ...
+
```
Last three paragraphs are great!

ejolysmith commented on _blog/why-null-models-crispr.md:
```diff
@@ -0,0 +1,83 @@
+
+## Where do skewed spacer abundance distributions come from?
+
+Many different research groups have observed the same phenomenon: when bacteria use their CRISPR immune system, some spacers end up being a lot more common in the population than others. A *lot* more common: in the data we analyzed for this project, the most common spacer sequence was *ten thousand* times more common than the least common sequence. Distributions like this are often called ''long- ...
```
Every sentence in this paragraph has a colon. I suggest breaking some up. First sentence could be something like “Many different research groups have observed that when bacterian use...” . You could probably get rid of the “: a long tail” at the end, it’s a little repetitive.

ejolysmith commented on _blog/why-null-models-crispr.md:
```diff
@@ -0,0 +1,83 @@
+<img src="/img/posts/why-null-models-crispr/CRISPR_diagram_v3.svg" alt="crispr_diagram" width="600px" style="padding:5px;">
+</p>
+
+## Where do skewed spacer abundance distributions come from?
```
Does this lead to a subtitle in the actual post? Otherwise the transition here can be made smoother.

mbonsma replied:
Yes, the ## gives a subheading, if that's what you mean.

ejolysmith replied:
Yea that's what I meant. Looks good then!

ejolysmith commented on _blog/why-null-models-crispr.md:
```diff
@@ -0,0 +1,83 @@
+<img src="/img/posts/why-null-models-crispr/banfield_data.svg" alt="banfield_data" width="400px" style="padding:5px;">
+</p>
+
+This study also found that certain regions of the phage genome were more likely to be a source for spacers, and combined with this skewed abundance distribution, they concluded that the high-abundance spacers were special: their sequence makes them more likely to be high abundance. This sounds like a perfectly reasonable explanation, and it's supported by other experiments. Here’s a figure from a ...
```
How is the first part of the first sentence any different from the last paragraph? After finishing this paragraph I understood, but at first I was a little confused. 

For the second experiment, is this for individual bacteria or for two populations? 

Did the first study do anything to figure out that certain regions of the phage genome were more likely to be a source for spacers? Or did they just get to their conclusion from the skewed distribution? If it’s the latter, I think it would be cool if you waited to talk about the next study until after your null model. That way you could show how their conclusion might not be right, but that then with another experiment we can rule out the null model. You can conclude (like you do) by saying that experiments that can be reproduced like this are rare, and that often we only have the distributions. Otherwise it's a little back and forth: here's a skewed distribution -> here is the underlying reason behind this particular distribution -> but another one exists -> but it doesn't really matter because we all ready know what the real explanation is -> but in biology there are many systems where you couldn't do this second kind of experiment. Do you see what I mean? I think it would work better like: here's a skewed distribution -> conclusion from first study -> but wait, there's another explanation (enter null model) -> if we do more experiments we can rule out the null model -> more experiments often not possible.

mbonsma replied:
Yeah, I see what you're saying, but I think the point I was trying to make is a bit different, that you can't take "B causes A" to mean "A is evidence of B". I'll rework it to try to explain a bit better!

ejolysmith commented on _blog/why-null-models-crispr.md:
```diff
@@ -0,0 +1,83 @@
+
+Amazingly, we found a strikingly broad spacer abundance distribution. The figure below compares experimental data on the left and our simulation data on the right. Our null model shows that you don't need special spacer sequences to find that some are very common and some are rare: you can get large differences in abundance purely because of randomness[^4].
+
+[^4]: How exactly does the randomness produce this broad distribution? In a nutshell, it happens because there are multiple random events all happening together. A random subset of bacteria acquire spacers at random times, and then the processes of bacterial growth and death (also random) magnify the initial randomness. Imagine two different spacer acquisition events: if one happens a little earl ...
```
I think you should add a link to your paper somewhere in the main text, some people will probably ignore the footnotes.

ejolysmith commented on _blog/why-null-models-crispr.md:
```diff
@@ -0,0 +1,83 @@
+
+Does this mean that the study that found high correlation between replicates was wrong to conclude that acquisition was biased? Not at all, in fact their study demonstrates that beyond a doubt! But here's the very important catch: our work shows that if you observe a broad distribution by itself, you *cannot* conclude that biased acquisition is the cause, because we know you don't need that ingre ...
+
+Why does this matter? In lab experiments like those described here, you can do multiple copies of the same experiment under the same conditions to prove a result. But the world is full of important and fascinating research areas where that isn't possible. Any study of bacteria in a natural environment is inherently unreproducible: you can never turn back the clock and do things the same way twice ...
```
You can take out the “And” at the begining of the last sentence. 

Great story and great blog! Made me want to read the paper (which I will do now haha). Remember that if you disagree with my comments or want to add something you can just reply to them and we can do another round. Reviewers are often wrong!

mbonsma replied:
Thanks for the great comments! Just pushed some changes.

ejolysmith commented on _blog/why-null-models-crispr.md:
```diff
@@ -0,0 +1,83 @@
+<img src="/img/posts/why-null-models-crispr/Heler2019.svg" alt="Heler2019" width="450px" style="padding:5px;">
+</p>
+
+So we're done, right? We know why some spacers are highly abundant and others are not, and if we see a broad abundance distribution we know the cause. Or do we? Remember where we started: to figure out what's going on when we observe something, we have to rule out other explanations. Could there be other explanations for a spacer abundance distribution that looks like this? 
```
Nice! This is very clear here, but wasn’t as much in the intro (see comments there).

mbonsma committed:
Edits from Euan's comments:
- small formatting: remove subtitle, fix date
- rework introduction to be more clear
- rework data and model section to add more detail

ejolysmith commented:
Looks good! I don't have write access so I can't merge the pull request. Jeremy or Matt, could one of you click the merge button? (or give me write access hehe? )

jbRothschild commented on _blog/why-null-models-crispr.md:
```diff
@@ -0,0 +1,84 @@
+
+Experimentalists wade through the murky waters of biology, hoping to tease out knowledge by controlling the things they can control and carefully observing what they cannot. In a well-designed experiment, the murky water is illuminated and an insight is reached by painstakingly ruling out other explanations. This might sound logical and systematic, but how do you know when you're done considering ...
+
+When there are many possible explanations for an experimental observation, theory can help. There are lots of ways to design a theoretical model of a biological system, and how you do it depends a lot on what you want to find out. On one extreme, very *mechanistic* models try to include every known biological detail to the highest accuracy possible - these models are often trying to calculate spe ...
```
I think I know what you mean when you say "they are wrong in a way that helps us understand ....", but maybe it's more accurate to say they hold some grain of truth? It's not the fact that it's wrong that lets us understand what's going on (your choice to keep it in, I think the way you wrote it makes for a very nice picture still!)

mbonsma replied:
Haha good point, but I would say that it's the deliberate ignoring of certain biological "truth" in these models (their wrongness) that makes them simple enough to understand!

jbRothschild replied:
I agree! Keep it as such, I think it's good phrasing to plant this little thought in peoples mind.

jbRothschild commented on _blog/why-null-models-crispr.md:
```diff
@@ -0,0 +1,84 @@
+
+Many different research groups have observed that when bacteria use their CRISPR immune system, some spacers end up being a lot more common in the population than others. A *lot* more common: in the data we analyzed for this project, the most common spacer sequence was *ten thousand* times more common than the least common sequence. Distributions like this are often called ''long-tailed'' because ...
+
+This figure shows the abundance of each spacer sequence in the experiment, and each colour is a different day in the 14-day experiment. Each sequence is sorted from most common to least common (left to right), and the y-axis shows the frequency of each sequence. For most days, the most common sequence at rank 1 has a frequency of 10<sup>-1</sup> = 10%, and the least common sequences have frequenc ...
```
"from most common to least common (left to right) on the x-axis" just for symmetry with bringing up the x axis.
I want to point out that these are all suggestions and are not meant to criticize your style of writing whatsoever! I don't want to come off as seeming to tell you what to write and how to write it! Hopefully this helps.

mbonsma replied:
Definitely appreciate all the comments!

jbRothschild commented on _blog/why-null-models-crispr.md:
```diff
@@ -0,0 +1,84 @@
+
+Amazingly, we found a strikingly broad spacer abundance distribution. The figure below compares experimental data on the left and our simulation data on the right. Our null model shows that you don't need special spacer sequences to find that some are very common and some are rare: you can get large differences in abundance purely because of randomness[^4].
+
+[^4]: How exactly does the randomness produce this broad distribution? In a nutshell, it happens because there are multiple random events all happening together. A random subset of bacteria acquire spacers at random times, and then the processes of bacterial growth and death (also random) magnify the initial randomness. Imagine two different spacer acquisition events: if one happens a little earl ...
```
This is a great footnote for non-experts! I think a blog post written all together about randomness could be really interesting... trying to generalize what we do and explaining it to a wide audience.

mbonsma replied:
Agree!

jbRothschild commented on _blog/why-null-models-crispr.md:
```diff
@@ -0,0 +1,84 @@
+<img src="/img/posts/why-null-models-crispr/model_diagram_simplified.png" alt="model" width="800px" style="padding:5px;">
+</p>
+
+Amazingly, we found a strikingly broad spacer abundance distribution. The figure below compares experimental data on the left and our simulation data on the right. Our null model shows that you don't need special spacer sequences to find that some are very common and some are rare: you can get large differences in abundance purely because of randomness[^4].
```
Maybe explain what you mean by broad to non-experts? I like how you describe 'long-tail' earlier... along those lines?

jbRothschild commented on _blog/why-null-models-crispr.md:
```diff
@@ -0,0 +1,84 @@
+
+Does this mean that the study that found high correlation between replicates was wrong to conclude that acquisition was biased? Not at all, in fact their study demonstrates that beyond a doubt! But here's the very important catch: our work shows that if you observe a broad distribution by itself, you *cannot* conclude that biased acquisition is the cause, because we know you don't need that ingre ...
+
+Why does this matter? In lab experiments like those described here, you can do multiple copies of the same experiment under the same conditions to prove a result. But the world is full of important and fascinating research areas where that isn't possible. Any study of bacteria in a natural environment is inherently irreproducible: you can never turn back the clock and do things the same way twice ...
```
What is the 'it' referred to at the end of the paragraph?

mbonsma replied:
Removed "to find it" - I think that makes it more clear.

jbRothschild commented on _blog/why-null-models-crispr.md:
```diff
@@ -0,0 +1,84 @@
+<img src="/img/posts/why-null-models-crispr/banfield_data.svg" alt="banfield_data" width="400px" style="padding:5px;">
+</p>
+
+What can we learn from a spacer abundance distribution that looks like this? Does observing a broad spacer distribution tell us anything about the underlying process that created it? There are many reasons why some spacers might be more common than others. Bacteria might prefer to acquire certain spacer sequences over others, perhaps because of the way the phage genome is arranged or because of t ...
```
I like the framing in terms of A causes B and is B evidence of A! My brain follows that.

jbRothschild commented on _blog/why-null-models-crispr.md:
```diff
@@ -0,0 +1,84 @@
+
+What can we learn from a spacer abundance distribution that looks like this? Does observing a broad spacer distribution tell us anything about the underlying process that created it? There are many reasons why some spacers might be more common than others. Bacteria might prefer to acquire certain spacer sequences over others, perhaps because of the way the phage genome is arranged or because of t ...
+
+Experiments can chip away at the first question. In the original study, they found in addition to this broad distribution that certain regions of the phage genome were more likely to be a source for spacers. Based on this, they concluded that the high-abundance spacers were special: their sequence makes them more likely to be high abundance. That's one piece of evidence that spacer sequences matt ...
```
I think this looks like a tough paragraph but you've written it well. A does cause B.

jbRothschild commented on _blog/why-null-models-crispr.md:
```diff
@@ -0,0 +1,84 @@
+<img src="/img/posts/why-null-models-crispr/Heler2019.svg" alt="Heler2019" width="450px" style="padding:5px;">
+</p>
+
+So we're done, right? We know why some spacers are highly abundant and others are not, and if we see a broad abundance distribution we know the cause. Or do we? We've answered the question "does process A cause result B", but what about the other direction? Remember where we started: to figure out what's going on when we observe something, we have to rule out other explanations. Could there be ot ...
```
I guess you haven't used it in these words, but in layman's terms people might recognize this as correlation vs causation. Don't know if this needs to be included anywhere, I think you're clear enough.

mbonsma replied:
Oh yeah, that's a great point! I'll leave that out for now, but you've got me thinking!

jbRothschild reviewed (commented):
I think these are all minor comments. Great post, ready to merge when you give me the okay.

mbonsma committed:
add edits from Jeremy

mbonsma commented:
Made revisions from @jbRothschild's comments - we're good to merge! Thanks @ejolysmith as well!

mbonsma committed:
add links to papers cited
